addons/payment_alipay/controllers/main.py

```diff
@@ -14,54 +14,33 @@
 
 class AlipayController(http.Controller):
     _notify_url = '/payment/alipay/notify'  # TODO ARJ simple nitpicking, but wouldn't it be cleaner to make those constants?
-    _return_url = '/payment/alipay/return'
-
-    def _alipay_validate_data(self, **post):
-        resp = post.get('trade_status')  # TODO ARJ strange choice of var name for a payment status
-        if resp:
-            # TODO ARJ all of this is already done in _process_feedback_data
-            if resp in ['TRADE_FINISHED', 'TRADE_SUCCESS']:
-                _logger.info('Alipay: validated data')
-            elif resp == 'TRADE_CLOSED':
-                _logger.warning('Alipay: payment refunded to user and closed the transaction')
-            else:
-                _logger.warning('Alipay: unrecognized alipay answer, received %s instead of TRADE_FINISHED/TRADE_SUCCESS and TRADE_CLOSED' % (post['trade_status']))
-        if post.get('out_trade_no') and post.get('trade_no'):
-            # TODO ARJ wtf? do we really need to do this?
-            post['reference'] = request.env['payment.transaction'].sudo().search([('reference', '=', post['out_trade_no'])]).reference
-            return request.env['payment.transaction'].sudo()._handle_feedback_data('alipay', post)
-        return False  # TODO ARJ useless rtype bool, could be None
+    _return_url = '/payment/alipay/return'  # If these are constant, they should be modified in all modules
```

Comment:
@AntoineVDV I think it's quite clear like that. We need to change it everywhere if we change the convention. Up to you.

Reply:
Watch out, you're on the odoo/odoo PR. I'll close it to keep all discussions on the other repo.

About my comment: As it said, it was really nitpicking so there is no need to change it. Nonetheless, I don't believe that there is a convention for constants. Some are located in a `const.py` file, some are upper-case variables in the scope of the file, while here (and in other places) they are private class attributes.